src/graaflib/algorithm/shortest_path.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include <graaflib/graph.h>
+#include <graaflib/types.h>
+
+#include <concepts>
+#include <vector>
+
+namespace graaf::algorithm {
+
+enum class edge_strategy { WEIGHTED, UNWEIGHTED };
+
+template <typename E>
+struct GraphPath {
+  std::vector<vertex_id_t> vertices;
```

Comment:
An `std::list` would also be suitable here instead of the `std::vector`. Since it is implemented as a doubly-linked list you could save having to `reverse` it after rebuilding the path.